### PR TITLE
Use the more informative generic type inference failure error on method calls on raw pointers

### PR DESCRIPTION
**Pull request by oli-obk**

### CODE REVIEW

**Review by WaffleLapkin — APPROVED** [review]
r=me with green CI

**Comment by GuillaumeGomez** [comment]
You're not supposed to remove long error explanation files. Take a look at `E0001.md` to see how to update it.

File: compiler/rustc_error_codes/src/error_codes/E0699.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#### Note: this error code is no longer emitted by the compiler.
+
```
Comment (GuillaumeGomez) [inline compiler/rustc_error_codes/src/error_codes/E0699.md:2]: You also need to remove `E0699` from the failing code example attributes just below.

File: compiler/rustc_error_codes/src/lib.rs [hunk 2]
```diff
@@ -441,7 +441,7 @@ E0695: 0695,
 E0696: 0696,
 E0697: 0697,
 E0698: 0698,
-E0699: 0699,
+E0699: 0699, // REMOVED: merged into generic inference var error
```
Comment (GuillaumeGomez) [inline compiler/rustc_error_codes/src/lib.rs:444]: You can comment it and move it to the commented list below (but keep your comment).

Comment (oli-obk) [reply compiler/rustc_error_codes/src/lib.rs:444]: no I cannot 😆 I tried this first.

Comment (GuillaumeGomez) [reply compiler/rustc_error_codes/src/lib.rs:444]: Wait really? O.o

Dark magic going on...

Comment (oli-obk) [reply compiler/rustc_error_codes/src/lib.rs:444]: The comment at the top of the file says to not remove things anymore

Comment (GuillaumeGomez) [reply compiler/rustc_error_codes/src/lib.rs:444]: Hum... [E0744](https://github.com/rust-lang/rust/blob/master/compiler/rustc_error_codes/src/lib.rs#L677) is commented, its file [still exists](https://github.com/rust-lang/rust/blob/master/compiler/rustc_error_codes/src/error_codes/E0744.md) and you can get it from `rustc --explain E0744`, so I'm surprised. Oh well, as long as CI passes...

**Pull request merged**